src/Datastore/Entity.php

```diff
@@ -0,0 +1,333 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Datastore;
+
+use ArrayAccess;
+use InvalidArgumentException;
+use JsonSerializable;
```

Comment:
Curious on your thoughts on this:

Since users can now extend `Entity` do you think it is still safe to use `JsonSerializable`? I think if a user wanted to define a custom json serialization format it would conflict with what we have here and cause errors.